src/auto_scheduler/transform_step.cc

```diff
@@ -923,5 +958,275 @@ String ComputeRootStepNode::PrintAsPythonAPI(Array<te::Stage>* stages,
   return ss.str();
 }
 
+/********** Primitives adding new stages **********/
+
+/*!
+ * \brief Common part for steps that add new stages(e.g. CacheReadStep, CacheWriteStep,
+ * RfactorStep). This will return all steps that can change the number of stages in a ComputeDAG,
+ * and stop by the current step.
+ */
+Array<Step> GetFormerStageModifiableSteps(Step current_step, const Array<Step>& transform_steps) {
```

Comment:
This naming seems not quite clear. How about `GetMutableSteps`?

Reply:
Or `StageMutableSteps` ?
It seems that just `MutableSteps` is still not quite clear.